ddtrace/internal/agent.py

```diff
@@ -84,9 +67,114 @@ def get_stats_url():
     return url
 
 
+# This method returns if a hostname is an IPv6 address
+def is_ipv6_hostname(hostname):
+    # type: (Union[T, str]) -> bool
+    if not isinstance(hostname, str):
+        return False
+    try:
+        socket.inet_pton(socket.AF_INET6, hostname)
+        return True
+    except socket.error:  # not a valid address
+        return False
+
+
+class AgentConfig(DDConfig):
+    __prefix__ = "dd"
+
+    _trace_agent_hostname = DDConfig.v(
+        Optional[str],
+        "trace_agent_hostname",
+        default=None,
+        help_type="String",
+        help="Legacy configuration, stores the hostname of the trace agent",
+    )
+
+    _trace_agent_port = DDConfig.v(
+        Optional[int],
+        "trace_agent_port",
+        default=None,
+        help_type="Int",
+        help="Legacy configuration, stores the port of the trace agent",
+    )
+
+    _trace_agent_url = DDConfig.v(
+        Optional[str],
+        "trace_agent_url",
+        default=None,
+        help_type="String",
+        help="Stores the URL of the trace agent",
+    )
+
+    trace_agent_timeout_seconds = DDConfig.v(
+        float,
+        "trace_agent_timeout_seconds",
+        default=DEFAULT_TIMEOUT,
+        help_type="Float",
+        help="Stores the timeout in seconds for the trace agent",
+    )
+
+    _dogstatsd_host = DDConfig.v(
+        Optional[str],
+        "dogstatsd_host",
+        default=None,
+        help_type="String",
+        help="Stores the hostname of the agent receiving DogStatsD metrics",
+    )
+
+    _dogstatsd_port = DDConfig.v(
+        Optional[int],
+        "dogstatsd_port",
+        default=None,
+        help_type="Int",
+        help="Stores the port of the agent receiving DogStatsD metrics",
+    )
+
+    _dogstatsd_url = DDConfig.v(
+        Optional[str],
+        "dogstatsd_url",
+        default=None,
+        help_type="String",
+        help="Stores the URL of the DogStatsD agent",
+    )
+
+    _agent_host = DDConfig.v(
+        Optional[str],
+        "agent_host",
+        default=None,
+        help_type="String",
+        help="Stores the hostname of the agent",
+    )
+
+    _agent_port = DDConfig.v(
+        Optional[int],
+        "agent_port",
+        default=None,
+        help_type="Int",
+        help="Stores the port of the agent",
+    )
+    # Effective trace agent URL (this is the one that will be used)
+    trace_agent_url = DDConfig.d(str, _derive_trace_url)
+    # Effective DogStatsD URL (this is the one that will be used)
+    dogstatsd_url = DDConfig.d(str, _derive_stats_url)
+
+
+config = AgentConfig()
+
+
+def get_trace_url() -> str:
+    """Return the Agent URL computed from the environment."""
+    return config.trace_agent_url
+
+
+def get_stats_url() -> str:
+    """Return the DogStatsD URL computed from the environment."""
+    return config.dogstatsd_url
```

Comment:
If it doesn't take too many changes, we could try to remove these functions as well